### ISSUE TITLE
r.minval/r.maxval can be written to rethinkdb._debug_scratch

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#4032]
The `polyglot/mutation/insert` tests are failing inside `interface.artificial_table` because that table seems to avoid our check that `r.minval` and `r.maxval` are not insertable and so the checks for errors when inserting those into `rethinkdb._debug_scratch` are failing. @danielmewes guesses that this is due to the lack of serialization for rethinkdb._debug_scratch.

This is easily demonstrated outside of the tests:
```python
>>> r.db('rethinkdb').table('_debug_scratch').insert({'value':r.minval}).run()
{u'errors': 0, u'deleted': 0, u'generated_keys': [u'0b48f71a-c5e8-4831-aae0-9207332b76d7'], u'unchanged': 0, u'skipped': 0, u'replaced': 0, u'inserted': 1}
>>> r.db('rethinkdb').table('_debug_scratch').run()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "rethinkdb/ast.py", line 97, in run
    return c._start(self, **global_optargs)
  File "rethinkdb/net.py", line 520, in _start
    return self._instance.run_query(q, global_optargs.get('noreply', False))
  File "rethinkdb/net.py", line 400, in run_query
    raise res.make_error(query)
rethinkdb.errors.RqlRuntimeError: Cannot convert `r.minval` to JSON. in:
r.db('rethinkdb').table('_debug_scratch')
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
``` 

This is probably isolated to the `rethinkdb._debug_scratch` table, so it is unlikely that actual users would ever encounter it, but it would be nice to solve as it would be complicated to work around in the test.

**Comment by danielmewes** [comment]
A fix for this is in CR 3249 with @mlucy

**Comment by danielmewes** [comment]
This was fixed in next by 750f09c7a3ce66a1cb59b122cbd31b1a169981e4 .
I also cherry-picked the fix into v2.1.x 7362e1c, because the same issue existed there.

On next, the artificial table test is currently still failing because of #4669.
